builtin/common/misc_helpers.lua

```diff
@@ -159,6 +159,56 @@ function dump(o, indent, nested, level)
 end
 
 --------------------------------------------------------------------------------
+-- Allow indexing strings
+
+getmetatable"".__index = function(str, index)
+	local itype = type(index)
+	if itype == "string" then
+		return string[index] or str:find(index)
```

Comment:
I didn't mean on purpose, but something like:

```lua
if not string.empty then
  --- return if this is an empty string
  function string:empty() return self == "" end
end

-- later some other mod:
if bucket_name["empty"] …
```

Sure, this is a made up example, and maybe things never will break by that. But i wouldn't want to check every mod out there to maybe break on my addition.